back-end/aws-cloud9-api/src/main/java/com/serverless/dal/QuestionDBTable.java

```diff
@@ -0,0 +1,199 @@
+package com.serverless.dal;
+
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapperConfig;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBScanExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+@DynamoDBTable(tableName = "PLACEHOLDER_PRODUCTS_TABLE_NAME")
+public class QuestionDBTable {
+
+private static final String QUESTIONS_TABLE_NAME = System.getenv("QUESTIONS_TABLE_NAME");
+    
+	private static DynamoDBAdapter db_adapter;
+    private final AmazonDynamoDB client;
+    private final DynamoDBMapper mapper;
+
+
+
+    
+
+    
+    
+   
+
+	public QuestionDBTable() {
+	        // build the mapper config
+	        DynamoDBMapperConfig mapperConfig = DynamoDBMapperConfig.builder()
+	            .withTableNameOverride(new DynamoDBMapperConfig.TableNameOverride(QUESTIONS_TABLE_NAME))
+	            .build();
+	        // get the db adapter
+	        this.db_adapter = DynamoDBAdapter.getInstance();
+	        this.client = this.db_adapter.getDbClient();
+	        // create the mapper with config
+	        this.mapper = this.db_adapter.createDbMapper(mapperConfig);
+	}
+	
+	   public List<Question> list() throws IOException {
+	    	DynamoDBScanExpression scanExp = new DynamoDBScanExpression();
+	        List<Question> results = this.mapper.scan(Question.class, scanExp);
+	        return results;
+	    }
+	   
+	   
+	   public List<ClosedQuestion> listclosed() throws IOException {
+	    	DynamoDBScanExpression scanExp = new DynamoDBScanExpression();
+	        List<ClosedQuestion> results = this.mapper.scan(ClosedQuestion.class, scanExp);
+	        return results;
+	    }
```

Comment:
Można to uprościć. Te dwie funkcje różnią się jedynie zwracanym typem. Można tam wstawić typ parametryczny, a dla jakiego to będzie, decydować będzie typ przy wywołaniu funckji w handlerze. Powinno to wyglądać tak:
`public List<T> list() throws IOException {
	    	DynamoDBScanExpression scanExp = new DynamoDBScanExpression();
	        List<T> results = this.mapper.scan(T.class, scanExp);
	        return results;
	    }`